.travis.yml

```diff
@@ -0,0 +1,62 @@
+os: linux
+dist: xenial
+sudo: false
+
+language: c
+
+addons:
+  apt:
+    packages:
+     - build-essential
+     - perl
+     - curl
+
+compiler:
+  - gcc
+  - clang
+
+env:
+  global:
+    - JOBS=2
+    - WORK=$TRAVIS_BUILD_DIR/work
+    - PREFIX=$HOME/kopenresty
+    - LUAROCKS=3.1.3
+    - PCRE=8.43
+  matrix:
+    - OPENRESTY=1.11.2.5 OPENSSL=1.0.2s NO_OPENRESTY_PATCHES=1
+    - OPENRESTY=1.13.6.2 OPENSSL=1.0.2s NO_OPENRESTY_PATCHES=1
+    - OPENRESTY=1.13.6.2 OPENSSL=1.1.0k
+    - OPENRESTY=1.13.6.2 OPENSSL=1.1.1c
+
+stages:
+  - unit tests
+  - test
+
+jobs:
+ include:
+   - stage: unit tests
+     script: for f in t/*.sh; do bash $f; done
+     env:
+
+install:
+  - if [[ $NO_OPENRESTY_PATCHES == "1" ]]; then OPENRESTY_PATCHES=--no-openresty-patches; fi
+  - echo $OPENRESTY_PATCHES
+
+script:
+  - ./kong-ngx-build --work $WORK --prefix $PREFIX --jobs $JOBS --openresty $OPENRESTY --openssl $OPENSSL --luarocks $LUAROCKS --pcre $PCRE $OPENRESTY_PATCHES >build.log 2>&1 || (cat build.log; exit 1)
+  - export PATH=$PREFIX/openresty/bin:$PATH
+  - export PATH=$PREFIX/luarocks/bin:$PATH
+  - export PATH=$PREFIX/openssl/bin:$PATH
+  - (openresty -v 2>&1 | grep -F -- "$OPENRESTY") || (openresty -v; exit 1)
+  - (openssl version | grep -F -- "$OPENSSL") || (openssl version; exit 1)
+  - (luarocks --version | grep -F -- "$LUAROCKS") || (luarocks --version; exit 1)
+  - (ldd $(which openresty) | grep -E -- "libssl.*?$PREFIX") || (ldd $(which openresty); exit 1)
+  - (ldd $(which openresty) | grep -E -- "libcrypto.*?$PREFIX") || (ldd $(which openrest)y; exit 1)
+  - (ldd $(which openresty) | grep -E -- "libluajit.*?$PREFIX") || (ldd $(which openresty); exit 1)
+  - (openresty -V 2>&1 | grep -F -- "--with-pcre=$WORK/pcre-$PCRE") || (openresty -V; exit 1)
+  - if [[ $NO_OPENRESTY_PATCHES == "1" ]]; then [[ ! -f "$WORK/openresty-$OPENRESTY/bundle/.patch_applied" ]] || (echo "OpenResty patches mistakenly applied"; exit 1); fi
+  - if [[ $NO_OPENRESTY_PATCHES != "1" ]]; then [[ -f "$WORK/openresty-$OPENRESTY/bundle/.patch_applied" ]] || (echo "OpenResty patches not applied"; exit 1); fi
+  - openresty -V # for humans
```

Comment:
I like these tests A LOT thank you!

Could you add a `luarocks install version` (or any arbitrary package) just to ensure that `luarocks` knows how to locate `luajit`?

Reply:
If it couldn't locate LuaJIT, `luarocks --version` would already fail (or would have not been installed in the first place).

Reply:
Seems like when encountering the above, the command still exists with `0`. I added such cases (although without depending on installing modules, which feels like too much and too slow, as well as too unreliable - we've had spurious failures due to connectivity issues with LuaRocks/GitHub in our CI before).

Reply:
Those new tests just caught `DESTDIR` not being very nice:

https://travis-ci.com/Kong/openresty-build-tools/jobs/211662676#L417

So I just pushed yet another fix making `*_DESTDIR` variables more robust and less error-prompt for future maintainers as well.

Reply:
yeah @gszr and I hit this today which is what spurred me to ask for the test 👍